drivers/hmis/app/models/hmis/ce/match/expression/client_field_map.rb

```diff
@@ -34,77 +46,88 @@ def arel
 
     def all
       @all ||= {
-        veteran_status: {
-          instance_value: lambda(&:veteran_status),
-          arel_field: arel.c_t['VeteranStatus'],
-          format_for_display: ->(v) { HudUtility2026.veteran_status(v) },
-        },
-        current_age: {
-          instance_value: ->(c) { c.age(current_date) },
-          arel_field: age_from(current_date, arel.c_t['DOB']),
-        },
-        days_homeless: {
-          instance_value: ->(c) do
-            GrdaWarehouse::Hud::Client.days_homeless(client_id: c.id)
-          end,
-          format_for_display: ->(days) { days.nil? ? nil : "#{days} #{'day'.pluralize(days)}" },
-        },
-        # Array of Project Types at which the Client has an open Enrollment, including WIP enrollments.
-        open_enrollment_project_types: {
-          instance_value: ->(c) do
-            Hmis::Hud::Enrollment.joins(client: :warehouse_client_source).
-              where(warehouse_clients: { destination_id: c.id }).
-              open_including_wip.
-              joins(:project).
-              distinct.
-              pluck(arel.p_t['ProjectType'])
-          end,
-          format_for_display: method(:map_project_types),
-        },
-        # Array of Project Types at which the Client has an open Enrollment, excluding WIP enrollments.
-        open_enrollment_project_types_excluding_incomplete: {
-          instance_value: ->(c) do
-            Hmis::Hud::Enrollment.joins(client: :warehouse_client_source).
-              where(warehouse_clients: { destination_id: c.id }).
-              open_excluding_wip.
-              joins(:project).
-              distinct.
-              pluck(arel.p_t['ProjectType'])
-          end,
-          format_for_display: method(:map_project_types),
-        },
-        # Array of Project Types at which the Client has an active Referral (e.g. not yet declined or accepted)
-        open_referral_project_types: {
-          instance_value: ->(c) do
-            Hmis::Ce::Referral.joins(client: :warehouse_client_source).
-              where(warehouse_clients: { destination_id: c.id }).
-              active.
-              joins(:target_project).
-              distinct.
-              pluck(arel.p_t['ProjectType'])
-          end,
-          format_for_display: method(:map_project_types),
-        },
+        days_since_last_exit: days_since_last_exit_field,
+        veteran_status: veteran_status_field,
+        current_age: current_age_field,
+        days_homeless: days_homeless_field,
+        open_enrollment_project_types: open_enrollment_project_types_field,
+        open_enrollment_project_types_excluding_incomplete: open_enrollment_project_types_excluding_incomplete_field,
+        open_referral_project_types: open_referral_project_types_field,
+      }
+    end
+
+    private
+
+    def days_since_last_exit_field
```

Comment:
This looks good! Am I correct in understanding:
* "days since last exit", "veteran status", and "current age" are currently the only fields that affect pre-filtering, since they have `arel_field` defined
* CDEs fields and open_enrollment_project_types are now evaluate on a _batch_ of clients instead of an individual client. Previously they were N+1 where N is the number of clients being evaluated, and now they are N+1 where N is the number of batches being evaluated